CHANGELOG.md

```diff
@@ -3,11 +3,23 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Development
+
+### `Changed`
+
+- Modified the template for input csv file to include a `sample_name` column in addition to `sample` in-line with changes to [IRIDA-Next update] as seen with the [speciesabundance pipeline]
+  - `sample_name` special characters will be replaced with `"-"`
+  - If no `sample_name` is supplied in the column `sample` will be used
+  - To avoid repeat values for `sample_name` all `sample_name` values will be suffixed with the index of the `input` samplesheet.csv
```

Comment:
I'm unclear on this one (and maybe behind on the conversation), but is this suffixing desired? Can we use the `sample` as the primary key, or was this already decided that we were using `sample_name` as the primary key because it was easier?

Reply:
That's fair! I started the conversation in the [speciesabundances PR](https://github.com/phac-nml/speciesabundance/pull/24). I just went with my idea here. We can decide on proper solution as a team.

As to why we are using `sample_name` as the primary key. We're using `sample_name` to address a couple IRIDA-Next GSP requests:
- Sample name, not IRIDA-Next ID, to be in the file outputs, or in the filenames.
- Put the sample name in the input samplesheet so it can be easily identified what IRIDA-Next will be using

This work was originally taken on by Kyla in the PR up-top. 

As to the suffixing, we need the `sample_name` to be unique even if users repeat the `sample_name`, and because we're no longer relying on IRIDA-Next autogenerated unique ID, we need to put in place a way to make it unique for Nextflow. Kyla originally proposed using a combination of `sample` and `sample_name`, I just wanted to try out an alternative idea.

Reply:
After discussing a bit more with @apetkau I believe the consensus is to go with the original solution preposed by @Kyla in the `speciesabundance` PR. Thanks to everyone for indulging me in my suggestion! I will modify the code unless there are any objections.

Reply:
Thanks so much, both for looking into different ways of uniquely naming samples @sgsutcliffe and for the great discussion on the best approach 😄. It's really great to think these things through. Including the IRIDA Next id as a suffix is the way to go forward.